checkstyle-tester/checks-nonjavadoc-error.xml

```diff
@@ -139,10 +139,6 @@
       <property name="tokens" value="ANNOTATION_FIELD_DEF"/>
       <property name="allowSamelineSingleParameterlessAnnotation" value="false"/>
     </module>
-    <module name="AnnotationLocation">
-      <property name="tokens" value="PARAMETER_DEF"/>
-      <property name="allowSamelineMultipleAnnotations" value="true"/>
```

Comment:
http://checkstyle.sourceforge.net/config_annotation.html#AnnotationLocation_Properties
Default value is `false`. I think we should leave the module in there and just remove the token so this property is still tested.

Reply:
With which tokens?
Default?
Do you think it will then work properly?

Reply:
Yes, just remove tokens so it runs with default and keep allowSamelineMultipleAnnotations to true.
It shouldn't be an issue, older versions will still work as it is the loading of the check in `TreeWalker` that defines what tokens will be used for default.

Reply:
I just had concerns that then checks are run that were not run before and could cause breakage (as all default tokens are tested instead of just the parameter def), but if you say so, I'm fine, fixed.

Reply:
If there is any breakage, then it will show us another problem in the check.
Checks should not throw an exception with any optional tokens added/removed. If a token is required for the check to work correctly, then it should be listed as required.

Reply:
I more thought about new violations found that are not expected.
But maybe I misinterpreted the intention of these regression checks, I didn't look closer into them.

Reply:
This specific check is for non-exception testing. 
Severity is set to ignore, so no violations are printed except exceptions.
https://github.com/checkstyle/contribution/blob/46221abc3b446b0826d363853cccb2028d16cadc/checkstyle-tester/checks-nonjavadoc-error.xml#L17

When we use project's own config in regression, it is violation testing. So we expect no new violations as long as they are strict in using checkstyle before adding new commits to their own repo.

Reply:
Ah, ok, that makes sense, thanks.